### DISCUSSION TITLE
Blazor WebAssembly support for ONNX Runtime Web

### DISCUSSION THREAD

**Discussion by jdluzen** [onnx/onnx#4942] [Q&A] [answered]
With ONNX Runtime Web compiled to wasm, I had hoped that I could add `Microsoft.ML.OnnxRuntime.Managed` to a Blazor project, include the ORW script tags, and it work through the existing pinvoke support. This does not seem to be the case. Is this,  will this be, or is it planned to be, a supported scenario? The alternative seems to be to go through the JS interop instead.

**Comment by cassiebreviu** [comment, marked as answer]
Hey! Here are some steps to get it to work:

Right now, you have to:
- [Build the ORT static lib](https://onnxruntime.ai/docs/build/web.html#build-instructions).
     - `./build.sh --build_wasm --build_wasm_static_lib --skip_tests --enable_wasm_debug_info`
- Rename file to onnxruntime.a
- [Import native reference in csproj file](https://learn.microsoft.com/en-us/aspnet/core/blazor/webassembly-native-dependencies?view=aspnetcore-7.0)

Check out this discussion post for more info: https://github.com/microsoft/onnxruntime/discussions/14657.
